src/pal/src/thread/process.cpp

```diff
@@ -1643,24 +1771,38 @@ class PAL_RuntimeStartupHelper
 
             for (ProcessModules *entry = listHead; entry != NULL; entry = entry->Next)
             {
-                if (IsCoreClrModule(entry->Name))
+                pe = IsCoreClrProcessReady(entry, pCoreClrIsReady);
+                if (pe != NO_ERROR)
+                {
+                    goto exit;
+                }
+
+                if (*pCoreClrIsReady)
                 {
                     PAL_CPP_TRY
                     {
                         TRACE("InvokeStartupCallback executing callback %p %s\n", entry->BaseAddress, entry->Name);
                         m_callback(entry->Name, entry->BaseAddress, m_parameter);
                     }
-                        PAL_CPP_CATCH_ALL
+                    PAL_CPP_CATCH_ALL
                     {
                     }
                     PAL_CPP_ENDTRY
 
-                        // Currently only the first coreclr module in a process is supported
-                        break;
+                    // Currently only the first coreclr module in a process is supported
+                    break;
                 }
             }
 
         exit:
+            if (*pCoreClrIsReady)
```

Comment:
Previously setting the continue semaphore was done unconditionally in StartupHelperThread and I wasn't sure what the rational was to move it here instead. I don't think we want to make setting this semaphore conditional on m_canceled right?